code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -500,16 +490,14 @@ func (uar *updateAllocationRequest) decode(b []byte) error {
 func (uar *updateAllocationRequest) validate(
 	conf *Config,
 	alloc *StorageAllocation,
-) (err error) {
+) error {
 	if uar.SetImmutable && alloc.IsImmutable {
 		return errors.New("allocation is already immutable")
 	}
-
-	if uar.Size == 0 && uar.Expiration == 0 {
+	if uar.Size == 0 && uar.Expiration == 0 && len(uar.AddedBlobberId) == 0 {
```

Comment:
What if uar.RemovedBlobberId is not empty

Reply:
We cannot reduce blobber count. There i[s a check a few lines down ](https://github.com/0chain/0chain/blob/update-blobbers/code/go/0chain.net/smartcontract/storagesc/allocation.go#L518)that checks if we remove a blobber we also add one. The error message here is more explicit.